pkg/volume/util/fs/fs.go

```diff
@@ -55,8 +56,29 @@ func FsInfo(path string) (int64, int64, int64, int64, int64, int64, error) {
 	return available, capacity, usage, inodes, inodesFree, inodesUsed, nil
 }
 
+// hangingDuCheck will stop executing DiskUsage() if this check is pending from last period.
+func pendingDuCheck(path string) error {
+	out, err := exec.Command("bash", "-c", fmt.Sprintf("ps aux | grep 'nice -n 19 du -x -s -B 1 %s' | wc -l", path)).Output()
```

Comment:
I'm actually a little confused about how we can get into the state where there are lots of du's. 

The volume metrics are [gathered serially](https://github.com/kubernetes/kubernetes/blob/a6f209c1c038dce7256e1cf115ed393ec22fbb9e/pkg/kubelet/server/stats/volume_stat_calculator.go#L108), and the call to [JitterUntil](https://github.com/kubernetes/kubernetes/blob/a6f209c1c038dce7256e1cf115ed393ec22fbb9e/pkg/kubelet/server/stats/volume_stat_calculator.go#L64) is supposed to wait until the previous one completed.

/assign @gnufied 
to also help take a look

Reply:
Sorry, I have one suspicion:
if we want collect it serially, does [JitterUntil](https://github.com/kubernetes/kubernetes/blob/a6f209c1c038dce7256e1cf115ed393ec22fbb9e/pkg/kubelet/server/stats/volume_stat_calculator.go#L64) and [Forever](https://github.com/kubernetes/kubernetes/blob/a6f209c1c038dce7256e1cf115ed393ec22fbb9e/pkg/kubelet/server/stats/fs_resource_analyzer.go#L65) duplicated?

Reply:
We appear to be spinning one goroutine for each Pod on the node and that is what `Forever` does. But metric collection for each pod should be happening concurrently.  @gongguan which version of kube you are running?

Also - I am bit uncomfortable about using `bash` and random mix of `ps aux`. I know there are minor incompatiblities between versions.

Reply:
> @gongguan which version of kube you are running?

I have met it in release-1.9 more than one year, [#88767](https://github.com/kubernetes/kubernetes/issues/88767) remind me of that. However version haven't been mentioned in the issue.

> Also - I am bit uncomfortable about using `bash` and random mix of `ps aux`. I know there are minor incompatiblities between versions.

Actually, I haven't found a very good solution yet.